### PR TITLE
Visualization Tools: Keys and Labels

### PR DESCRIPTION
**Pull request by haleyhowe**
This is the very beginning. Still need to implement different charts and also tweak inputs for visualization features/parameters within each plot.

Implementing Issue #9

### CODE REVIEW

**Comment by haleyhowe** [comment]
Please excuse my commit message about Issue #11, this was supposed to be Issue #9.
Recommitted with Issue #9 in commit message.

**Comment by rohitchadaram** [comment]
Hi Haley, As i see I think you have 3 files committed which might be the earlier version of the code you wanted to commit. You could remove those files by checking your git status and removing those files and committing, so that once you merge with the master you don't push those additional duplicate files.

File: hs628_tools/Project1_Vizualization.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+import pandas as pd
+import sys 
+import plotly.plotly as py
+import plotly.graph_objs as go
+#plotly.tools.set_credentials_file(username='hhowell', api_key='BRP1aHhfTHwkqnvalVQL')
```
Comment (rohitchadaram) [inline hs628_tools/Project1_Vizualization.py]: should never have the credentials displayed in your code, you can delete this line.

File: hs628_tools/Project1_Vizualization.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+###This is importing a class from Issue #11 Test module. We used the ratio function to determine the test size ratio for the train_test_split function. 
+#from Test import data_split as ds
+
+
+###Bar Plot-Count
+#This function counts all instances in a variable and creates a barplot to display the results 
+#@param: The variable to be counted, the x_axis label, title of the graph
+#@return: A count bar plot of the variable inputted 
+def count_and_print(variable, x_label, title):
+    def count(variable):
```
Comment (rohitchadaram) [inline hs628_tools/Project1_Vizualization.py]: Why do you have a function inside a function. I believe it is a good practice to call a function from another function but not define the function altogether. Is there are good reason which I am missing ?

File: hs628_tools/Project1_Vizualization.py [hunk 3]
```diff
@@ -159,48 +154,144 @@ def boxplot(x,y, y_label, x_label, title):
...
+
+Xtrain, Xtest, ytrain, ytest = train_test_split(X,Y, test_size = .5) #This assumes that Issue #11 Test.py module works and outputs the test ratio. 
+Xtrain = np.array(Xtrain)
+ytrain = np.array(ytrain)
+Xtest = np.array(Xtest)
+ytest = np.array(ytest)
+
+Q1= input("What type of plot output? (Enter #)"+"\n"+"\n"+"1.Countplot"+"\n"+"2.Scatterplot"+"\n"+"3.Boxplot"+"\n"+"4.Simple Linear Regression Model"+"\n"+"5.Multiple Linear Regression Model"+"\n")
+question(data,Xtrain,Xtest,ytrain,ytest, Q1)
+
```
Comment (RoxanneXin) [inline hs628_tools/Project1_Vizualization.py]: I really like the solution you and @NikitaThomas found out for this great idea. And I like the idea that  you use the the module train_test_split to get four parameters Xtrain, Xtest, ytrain, and ytest.

File: hs628_tools/Project1_Vizualization.py (hunk 1 above)
Comment (haleyhowe) [reply hs628_tools/Project1_Vizualization.py]: okay thanks Rohit , appreciate it!

File: hs628_tools/Project1_Vizualization.py (hunk 2 above)
Comment (haleyhowe) [reply hs628_tools/Project1_Vizualization.py]: There is not specific reason. I used this function within this function itself and only this function therefore I kept it in there. Not sure if this is a thing or not however I felt it was not a problem. Let me know!

**Comment by EHWUSF** [comment]
OK, well, it looks like the state as of your commit 7c2f00fe is *_largely_* identical to the state of Nikita's branch after her final commit, except that you wound up with more extra files, and your main file seems to have some extra superfluous lines in it.  So please refer to my comments on her PR with regards to the state up to that point.  

Your extra couple commits here are good additions/cleanup past that, and overall, it's important to have your branch/PR also, since right now each of your branches only reflects your own commit history plus the final state of the other's.  

Let's talk either during or after class about the best way for the two of you to proceed with cleanup and preserve the git history.
